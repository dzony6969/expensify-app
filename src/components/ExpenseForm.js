import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates'; 
import 'react-dates/lib/css/_datepicker.css' //styl kalendrza 

export default class ExpenseForm extends React.Component {
    constructor(props) {
        super(props)
        this.state = { 
        description: props.expense ? props.expense.description : '',
        note: props.expense ? props.expense.note : '',
        amount: props.expense ? (props.expense.amount /100).toString() : '',
        createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
        calendarFocused: false,
        error: ''

        }
    }
    OnDescriptionChange = (e) => {
        const description = e.target.value
        this.setState(() =>({description}));
        
    }
    onNoteChange = (e) => {
        e.persist() //inne wyjscie
        this.setState(() => ({note: e.target.value}))
    };
    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) { //regular expression 31233123.22 i nie wiecej
        this.setState(() => ({amount}))
        }
    };
    onDateChange = (createdAt) => {
        if(createdAt) {
        this.setState(() => ({createdAt}));
        }
    };
    onFocusChange = ({focused}) => {
        this.setState(() => ({calendarFocused: focused}));
    };
    onSubmit = (e) => {
        e.preventDefault();

        if(!this.state.description || !this.state.amount) {
          this.setState(() => ({error: 'Nie dodano nazwy oraz kwoty wydatku'}))
        } else {
            this.setState(() => ({error: ''}));
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note

            })
        }

    }

    render() {
        return (

       
              
               <form className='form' onSubmit={this.onSubmit}>
               {this.state.error && <p className='form__error'>{this.state.error}</p>}
                   <input 
                   type='text'
                   className='text-input'
                   placeholder='Nazwa wydatku'
                   autoFocus //po przelaczeniu od razu zaznacza input
                   value={this.state.description}
                   onChange={this.OnDescriptionChange}
                   />
                   <input type='text'
                   placeholder='Kwota'
                   className='text-input'
                   value={this.state.amount}
                   onChange={this.onAmountChange}
                   
                   />
                   <SingleDatePicker
                    date={this.state.createdAt}
                    onDateChange={this.onDateChange}
                    focused={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={() => false }
                    />
                   <textarea 
                   className='textarea'
                   placeholder='opis wydatku'
                   value={this.state.note}
                   onChange={this.onNoteChange}
                   >
                    </textarea>
                    <div>
                   <button className='button'>Wy≈õlij</button>
                   </div>
                </form>
                
         
        )
    }
}