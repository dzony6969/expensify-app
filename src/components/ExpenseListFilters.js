import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }
  render() {
    return (
      <div className='content-container'>
      <div className='input-group'>
      <div className='input-group__item'>
      <input
          placeholder='Szukaj'
          type="text"
          className='text-input'
          value={this.props.filters.text}
          onChange={(e) => {
            this.props.dispatch(setTextFilter(e.target.value));
          }}
        />

      </div>
      <div className='input-group__item'>
      <label>Filtry</label>
      <select
        className='select'
          value={this.props.filters.sortBy}
          onChange={(e) => {
            if (e.target.value === 'date') {
              this.props.dispatch(sortByDate());
            } else if (e.target.value === 'amount') {
              this.props.dispatch(sortByAmount());
            }
          }}
        >
          <option value="date">Data</option>
          <option value="amount">Kwota</option>
        </select>

      </div>
      <div className='input-group__item'>
      <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          showClearDates={true} //pokazuje X
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
        </div>
        
     
      </div>
    );
  }
};

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
