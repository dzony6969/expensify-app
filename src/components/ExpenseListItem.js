import React from 'react'
import { Link } from 'react-router-dom';
import moment from 'moment'
import numeral from 'numeral'




const ExpenseListItem = ({description, amount, createdAt, id }) => (
    <div>
     <Link to={`/edit/${id}`} >
      <h2>{description}</h2>
      </Link>
      <p> 
    {numeral(amount /100).format('0,0.00') + 'z≈Ç '}
       - 
    {moment(createdAt).format('Do MMMM, YYYY')}
      
      </p>
      
    </div>
);

export default (ExpenseListItem);
