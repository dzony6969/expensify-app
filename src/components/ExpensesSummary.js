import React from 'react'
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses'
import { connect } from 'react-redux';
import numeral from 'numeral';
import { Link } from 'react-router-dom';

const ExpensesSummary = ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal /100).format('0,0.00') + ('zł')
    return (
    <div className='page-header'>
    <div className='content-container'>
        <h1 className='page-header__title'>Oglądasz <span>{expenseCount}</span>  swoje wydatki <br/>  Suma: <span>{formattedExpensesTotal}</span></h1>
        <div className='page-header__actions'>
            <Link to='/create' className='button'> Dodaj nowy wydatek </Link>
        </div>
        </div>
    </div>
)
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)

    };
}


export default connect(mapStateToProps)(ExpensesSummary)