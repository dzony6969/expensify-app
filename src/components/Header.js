import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensive</h1>
        <NavLink to='/' activeClassName='is-active' exact={true}>Home page</NavLink>
        <NavLink to='/create' activeClassName='is-active'>Create new expensive</NavLink>
        <NavLink to='/help' activeClassName='is-active'>Sort</NavLink>
    </header>
)

export default Header;