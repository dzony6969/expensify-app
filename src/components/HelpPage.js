import React from 'react';

const HelpPage = () => (
    <div>
        From help page
    </div>
)

export default HelpPage;