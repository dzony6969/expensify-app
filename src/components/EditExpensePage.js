import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { setEditExpense, setRemoveExpense } from '../actions/expenses';
const EditExpensePage = (props) => {
    return (
    <div>
        edditing the expense of {props.match.params.id} 
          {/* pokazuje id w ktorym jestesmy */}
        <ExpenseForm 
        expense={props.expense}

        onSubmit={(expense) => {
            props.dispatch(setEditExpense(props.expense.id, expense)); //dodawaniee
            props.history.push('/');
        }}
        />
        <button onClick={(expense) => {
       props.dispatch(setRemoveExpense({id: props.expense.id}));
       props.history.push('/')
        // HandleDeleteItem musisz podac id i dispatch w w.w obiekcie
      }}>Remove</button>
    </div>
)
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find( expense => expense.id === props.match.params.id)
    }
}

export default connect(mapStateToProps)(EditExpensePage);