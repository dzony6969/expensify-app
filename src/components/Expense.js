import React from 'react';
import ExpenseList from './expenseList';
import ExpenseListFilters from './ExpenseListFilters';
const Expense = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export default Expense;