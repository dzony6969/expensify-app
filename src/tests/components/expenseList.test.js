import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/expenseList'
import expenses from '../fixtures/expenses';

test('should render expenseList with expenses', () =>{
    const wrapper = shallow(<ExpenseList expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseList with empy msg', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>)
    expect(wrapper).toMatchSnapshot();
})