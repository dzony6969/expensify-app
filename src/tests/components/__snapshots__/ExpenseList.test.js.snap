// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with empy msg 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;

exports[`should render expenseList with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={123}
    createdAt={0}
    description="gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={10837}
    createdAt={-345600000}
    description="rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="credit card"
    id="3"
    key="3"
    note=""
  />
</div>
`;
